### DISCUSSION TITLE
How to change the the Manning coefficient in subcatchment ？

### DISCUSSION THREAD

**Discussion by ZhouNie96** [pyswmm/pyswmm#414] [Q&A]
I want to implement automatic calibration of the SWMM model, which includes the Manning coefficient. How can I change this parameter in the subcatchment ?

![image](https://user-images.githubusercontent.com/116863674/228779616-242f4f37-65b7-4ec9-a89c-373bceaa43b3.png)

**Comment by bemcdonnell** [comment]
@ZhouNie96, this sounds like a great feature request. We are coming up with a series of videos for YouTube on how one can contribute new features like this.  Would you be interested in helping us make this happen?

**Comment by bemcdonnell** [comment]
For now, you can pre-process the INP file by doing a find-and-replace.

**Comment by Utsav1234567** [comment]
I am having the same problem. How to fix it ?

**Comment by bemcdonnell** [comment]
@Utsav1234567 I'm just taking my email to you here:

I am looking into the code right now and I agree with your discovery.   The N value has not been exposed through the API yet.  Furthermore, it looks like the percent impervious has also been unhooked.  

here is the SWMM function that is called for "setting" values in SWMM. I see that we have a comment in the code:
```C
            case SM_FRACIMPERV:
                break; //Subcatch[index].fracImperv; break;
                // Cannot Open Function just yet.  Need
                // to adjust some internal functions to
                // ensure parameters are recalculated
                // = MIN(value, 100.0) / 100.0; break;
```
https://github.com/OpenWaterAnalytics/Stormwater-Management-Model/blob/8266e4434bc9c23218bbf18fe57b0f3523d414ee/src/solver/toolkit.c#L992-L1041

I think it would be useful to have these.  Would you be interested in helping us integrate these changes?  There is minimal C programming and a bit of Python.  I can also do it for you but I don't think I can make it a priority at this time.

**Reply by bemcdonnell** [reply]
https://github.com/OpenWaterAnalytics/pyswmm/discussions/422 linking here.
